.teamcity/_self/projects/WPComTests.kt

```diff
@@ -181,3 +182,56 @@ fun gutenbergBuildType(screenSize: String, buildUuid: String): BuildType {
 		}
 	}
 }
+
+private object VisualRegressionTests : BuildType({
+	name = "Visual Regression Tests"
+	description = "Runs visual regression tests"
+
+	artifactRules = """
+		reports => reports
+	""".trimIndent()
+
+	vcs {
+		root(Settings.WpCalypso)
+		cleanCheckout = true
+	}
+
+	steps {
+		bashNodeScript {
+			name = "Prepare environment"
+			scriptContent = """
+				export NODE_ENV="test"
+
+				# Install modules
+				${_self.yarn_install_cmd}
+			"""
+		}
+		bashNodeScript {
+			name = "Run e2e tests (desktop)"
+			scriptContent = """
+				set -x
+
+				# Decrypt config
+				openssl aes-256-cbc -md sha1 -d -in ./test/visual/config/encrypted.enc -out ./test/visual/config/local-test.json -k "%CONFIG_E2E_ENCRYPTION_KEY%"
+
+				# Run the test
+				yarn test-visual
+			""".trimIndent()
+		}
+		bashNodeScript {
+			name = "Collect results"
+			executionMode = BuildStep.ExecutionMode.RUN_ON_FAILURE
+			scriptContent = """
+				set -x
+
+				mkdir -p screenshots
+				find test/visual -type f -path '*/html_report/*' -print0 | xargs -r -0 mv -t reports
```

Comment:
Something we can play with in the future: TeamCity has the option to embed any HTML report in their UI: https://www.jetbrains.com/help/teamcity/including-third-party-reports-in-the-build-results.html